packages/utils/src/store/Store.ts

```diff
@@ -71,4 +100,41 @@ export class Store<State> {
       this.update({ ...this.state, [key]: value });
     }
   }
+
+  /**
+   * Returns a value from the store's state using a selector function.
+   * Used to subscribe to specific parts of the state.
+   * This methods causes a rerender whenever the selected state changes.
+   *
+   * @param key Key of the selector to use.
+   */
+  public useState<Key extends keyof Selectors>(key: Key): ReturnType<Selectors[Key]> {
+    if (!this.selectors) {
+      throw new Error('Base UI: selectors are required to call useState.');
+    }
+    // False positive
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    return useStore<State, ReturnType<Selectors[Key]>>(
+      this,
+      this.selectors[key] as (state: State) => ReturnType<Selectors[Key]>,
+    );
+  }
+
+  /**
+   * Wraps a function with `useEventCallback` to ensure it has a stable reference
+   * and assigns it to the context.
+   *
+   * @param key Key of the event callback. Must be a function in the context.
+   * @param fn Function to assign.
+   */
+  public useEventCallback<Key extends ContextFunctionKeys<Context>>(
```

Comment:
The naming makes it feel like it should return a function like a `useEventCallback()` call would do. Maybe I'd rework the name a bit - `store.useContextCallback()`?.